libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -262,6 +262,14 @@ class MapboxNavigation(
         tripSession.stop()
     }
 
+    /**
+     * Clears the location and route for the navigator.
+     */
+    fun resetTripSession() {
+        tripSession.route = null
+        navigator.reset()
```

Comment:
Digging up some history as we had a similar function in the past.

Introduced a reset function, that would recreate the navigator https://github.com/mapbox/mapbox-navigation-android/pull/2850
That caused a regression in the examples https://github.com/mapbox/mapbox-navigation-android/issues/2769
Fixed the regression with https://github.com/mapbox/mapbox-navigation-android/pull/2905

Notice, the issue was because the reset function was called inside [MapboxTripSession.reset()](https://github.com/mapbox/mapbox-navigation-android/pull/2905/files/d48bb7ec1c23eeab1530de594bd978c67ec660a7#diff-f1923dcc227762ccf3eb5fcf1cab79bc5da009704c56901e0c78b6d257b5c1d6L187)

[But we're not calling it here](https://github.com/mapbox/mapbox-navigation-android/blob/aa333725a94a0cbd2edf7c79fd08879b8c5a35ad/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L205-L214)

Worth noting that, our MapboxTripSession private reset function is very different from the Nav Native's resetRideSession.

Reply:
I've also tested the repro steps included in the issue from before: https://github.com/mapbox/mapbox-navigation-android/issues/2769#issuecomment-616738902

And this change is looking good. It's because we don't want to add the Navigator.resetRideSession call to the MapboxTripSession.reset() function